granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -88,13 +198,46 @@ def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> O
     v2_path = find_v2_hierarchy(resolve_host_root_links)
     if v1_paths and controller in v1_paths:
         # Either v1 or hybrid with the requested controller bound to a v1 hierarchy
-        return v1_paths[controller]
+        return CgroupCoreV1(Path(v1_paths[controller]))
     if v2_path:
-        # v2 (unified) - all controllers are in a single unified hierarchy
-        return v2_path
+        # v2 (unified) - check given controller is supported in cgroup v2
+        cgroup_v2 = CgroupCoreV2(Path(v2_path))
+        if cgroup_v2.is_controller_supported(controller):
+            return cgroup_v2
     # No cgroup mount of the requested controller
     return None
 
 
-def is_known_controller(controller: str) -> bool:
-    return controller in SUBSYSTEMS
+def get_cgroup_mount_checked(controller: str, resolve_host_root_links: bool = True) -> CgroupCore:
+    cgroup_mount = get_cgroup_mount(controller)
+    assert cgroup_mount is not None, f"Could not find cgroup mount point for controller {controller!r}"
+    return cgroup_mount
+
+
+def create_cgroup_from_path(controller: str, cgroup_path: Path) -> CgroupCore:
+    cgroup_mount = get_cgroup_mount_checked(controller)
+
+    try:
+        cgroup_path.relative_to(cgroup_mount.path)
+    except ValueError:
+        cgroup_path = cgroup_mount.path / cgroup_path
+
+    if isinstance(cgroup_mount, CgroupCoreV1):
+        return CgroupCoreV1(cgroup_path)
+    else:
+        return CgroupCoreV2(cgroup_path)
+
+
+def get_cgroup_for_process(controller: str, process: Optional[psutil.Process] = None) -> CgroupCore:
+    """
+    Get a CgrouopCore object for a given process. If process is None return for current process.
+    """
+    cgroup_mount = get_cgroup_mount_checked(controller)
+    is_v1 = isinstance(cgroup_mount, CgroupCoreV1)
+
+    for process_cgroup in get_process_cgroups(process):
+        if is_v1 and controller in process_cgroup.controllers:
+            return CgroupCoreV1(cgroup_mount.path / process_cgroup.relative_path[1:])
+        elif not is_v1 and process_cgroup.hier_id == "0":
```

Comment:
Can `hier_id` ever be not 0 for v2?

If not - remove the `if` and instead `assert` it.